### PR TITLE
chore(apollo): upgrade apollo client dependencies

### PR DESCRIPTION
**Pull request by erikwrede**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #2911 

## What is the new behavior?
This PR fixes the issues described in #2911 by upgrading the outdated apollo client to a much newer version. The issue was not caused by an upstream bug, but by outdated dev dependencies incomatible with current prod dependencies.

### CODE REVIEW

File: packages/apollo/tests/subscriptions/graphql-ws.spec.ts [hunk 1]
```diff
@@ -37,7 +36,12 @@ describe('graphql-ws protocol', () => {
           },
           subscriptions: {
             'graphql-ws': {
-              onConnect: (context: Context<any>) => {
+              onConnect: (
+                context: Context<
+                  any,
+                  { socket: { close: (number, string) => any | boolean } }
+                >,
+              ) => {
```
Comment (erikwrede) [inline packages/apollo/tests/subscriptions/graphql-ws.spec.ts:44]: Did this to fix the typescript errors, if anyone can point me towards the right type that is used here, I'm happy to adjust this line.

**Comment by erikwrede** [comment]
@kamilmysliwiec any chance to get a review on this? 😊

**Comment by erikwrede** [comment]
@bel7aG since this is a dependency upgrade the package lock is changing a lot. 

If @kamilmysliwiec gives an okay I can update the lock **immediately**. Otherwise, there's little sense in updating the lock every week without feedback, given the frequent repo updates.

**Comment by kamilmysliwiec** [comment]
https://github.com/nestjs/graphql/pull/2704#issuecomment-1463730279

**Pull request closed without merging**